### PR TITLE
Sync v2-10-stable with v2-10-test to release 2.10.1rc1

### PR DESCRIPTION
**Pull request by utkarsharma2**
Time for 2.10.1rc1!

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,40 @@
...
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+No significant changes.
+
+Bug Fixes
+"""""""""
+- Handle Example dags case when checking for missing files (#41856)
+- logout link in no roles error page fix (#41813)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix logout link in "no roles" error page(#41813)
```

File: RELEASE_NOTES.rst [hunk 2]
```diff
@@ -21,6 +21,40 @@
...
+- Set end_date and duration for triggers completed with end_from_trigger as True. (#41834)
+- DAGs are not marked as stale if the dags folder change (#41433)
+- Keep FAB compatibility for versions before 1.3.0 in 2.10 (#41549)
+- Don't Fail LocalTaskJob on heartbeat (#41704)
+- Remove deprecation warning for cgitb in Plugins Manager (#41732)
+- Fix log for notifier(instance) without __name__ (#41591)
+- Splitting syspath preparation into stages (#41672)
```
Comment (ephraimbuddy) [inline RELEASE_NOTES.rst]:
```suggestion
- Splitting syspath preparation into stages (#41694)
```
Should point to the backport

**Review by ephraimbuddy — COMMENTED** [review]
Some of the generated changelog PR numbers are pointing to the main branch instead of v2-10-test branch.

File: RELEASE_NOTES.rst [hunk 3]
```diff
@@ -21,6 +21,43 @@
...
+- Keep FAB compatibility for versions before 1.3.0 in 2.10 (#41809)
+- Don't Fail LocalTaskJob on heartbeat (#41810)
+- Remove deprecation warning for cgitb in Plugins Manager (#41793)
+- Fix log for notifier(instance) without __name__ (#41699)
+- Splitting syspath preparation into stages (#41694)
+- Adding url sanitization for extra links (#41680)
+- Fix InletEventsAccessors type stub (#41607)
+- Fix UI rendering when XCom is INT, FLOAT, BOOL or NULL (#41605)
+- Fix try selector refresh (#41503)
+- Incorrect try number subtraction producing invalid span id for OTEL airflow (issue #41501) (#41535)
```
Comment (utkarsharma2) [inline RELEASE_NOTES.rst:47]:
```suggestion
- Incorrect try number subtraction producing invalid span id for OTEL airflow (#41535)
```

**Review by potiuk — APPROVED** [review]

File: RELEASE_NOTES.rst [hunk 4]
```diff
@@ -21,6 +21,43 @@
...
+
+No significant changes.
+
+Bug Fixes
+"""""""""
+- Handle Example dags case when checking for missing files (#41874)
+- Fix logout link in "no roles" error page (#41845)
+- Set end_date and duration for triggers completed with end_from_trigger as True. (#41834)
+- DAGs are not marked as stale if the dags folder change (#41829)
+- Keep FAB compatibility for versions before 1.3.0 in 2.10 (#41809)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]:
```suggestion
- Fix compatibility with FAB provider versions <1.3.0 (#41809)
```

File: RELEASE_NOTES.rst [hunk 5]
```diff
@@ -21,6 +21,43 @@
...
+- Add WebEncoder for trigger page rendering to avoid render failure (#41485)
+- Adding ``tojson`` filter to example_inlet_event_extra example dag (#41890)
+- Add backward compatibility check for executors that don't inherit BaseExecutor (#41927)
+
+Miscellaneous
+"""""""""""""
+- Bump webpack from 5.76.0 to 5.94.0 in /airflow/www (#41879)
+- Adding rel property to hyperlinks in logs (#41783)
+- Field Deletion Warning when editing Connections (#41504)
+- Remove Debian bullseye support (#41569)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: This is docs only for core. Which makes me wonder, there are no other docs-only changes for 2.10.1?

File: RELEASE_NOTES.rst [hunk 6]
```diff
@@ -21,6 +21,43 @@
 
 .. towncrier release notes start
 
+Airflow 2.10.1 (2024-08-30)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst]: Don't forget to update this date!

**Review by jedcunningham — APPROVED** [review]

File: RELEASE_NOTES.rst (hunk 6 above)
Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: Updated it to 2024-09-05, assuming we are doing rc1 today.

File: RELEASE_NOTES.rst (hunk 5 above)
Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: Moved it to the Docs section. Also, there was a [PR](https://github.com/apache/airflow/commit/83aca021edb9054f43451171c7f415d5a6205536) which was not picked at all by `./dev/airflow-github changelog v2-3-stable v2-3-test` command, that too was docs only change.

Comment (utkarsharma2) [reply RELEASE_NOTES.rst]: I have gone through other changes and I think we have two docs changes which I have listed under the Docs changes sections.

**Comment by potiuk** [comment]
If we have still the possibility #41939 might be also good to include.

**Review by ephraimbuddy — APPROVED** [review]

**Comment by ephraimbuddy** [comment]
Update the release note to include the recently added PRs

**Comment by utkarsharma2** [comment]
> Update the release note to include the recently added PRs

Sure, I'll update it.

**Review by phanikumv — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Looks green :)

**Pull request merged**